velox/exec/Window.cpp

```diff
@@ -32,11 +32,17 @@ Window::Window(
           "Window"),
       numInputColumns_(windowNode->sources()[0]->outputType()->size()),
       windowBuild_(std::make_unique<SortWindowBuild>(windowNode, pool())),
+      windowNode_(windowNode),
       currentPartition_(nullptr),
-      stringAllocator_(pool()) {
-  auto inputType = windowNode->sources()[0]->outputType();
-  createWindowFunctions(windowNode, inputType, driverCtx->queryConfig());
+      stringAllocator_(pool()) {}
 
+void Window::ensureInitialized() {
```

Comment:
I wonder if it makes sense to add a new initialize() API to Operator.

Reply:
Yeah, @tanjialiang plans to add this API can call it once for all the operators in a driver when we run that driver at the first time. Then it will do the initialization unconditionally even if the input is empty but it might be ok as it is not common. I try to catch all existing cases using https://github.com/facebookincubator/velox/pull/6409 first.